### PR TITLE
MediaModal component

### PR DESCRIPTION
**Pull request by nathanyoung**
# Github Issue or Trello Card
This PR addresses this issue: https://trello.com/c/0KKnSBL9/1133-create-component-for-media-asset-modal

A new component for displaying an image or video in a "light box" style container over existing page content.

Image only
Screen Shot 2022-03-14 at 12 51 42 PM

Image with Header and Footer
Screen Shot 2022-03-14 at 12 52 13 PM

Embedded vimeo video
Screen Shot 2022-03-14 at 12 52 39 PM

# What type of change is this?
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updating documentation.
- [ ] Updating deployment/build pipeline.
- [ ] Updating Project Dependencies
- [ ] Improving or adding to Test Coverage

# Completeness Checklist

- [x] TESTS: My changes maintain the baseline required test coverage, as specified by code climate analysis.
- [x] DOCS: All new component work is covered in that component's `Overview` docs.
- [x] DOCS: All new component work is covered in a component `Playground`.
- [x] DOCS: All new visual changes or options are covered under relevant components' `VisualTests`.
- [x] My code has no linting or typescript compile warnings.
- [x] My work is tied to a Github issue and satisfies the acceptance criteria (if applicable) of the corresponding issue.

# Quality Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# UI Checklist
- [ ] I have conducted visual UAT on my changes/features.
- [ ] My solution works well on desktop, tablet, and mobile browsers.

### CODE REVIEW

File: src/components/MediaModal/MediaModal.test.tsx
Comment (nathanyoung) [inline src/components/MediaModal/MediaModal.test.tsx:12]: Reusing our chromatic stories for unit testing

File: src/components/Modal/Modal.tsx
Comment (nathanyoung) [inline src/components/Modal/Modal.tsx:116]: Spelling

File: src/components/Modal/Modal.module.scss
Comment (nathanyoung) [inline src/components/Modal/Modal.module.scss:18]: this doesn't seem to impact the components themselves, but was there a reason we added this for users of `palmetto-components`?

File: src/components/MediaModal/MediaModal.test.tsx
Comment (juanfabrega) [reply src/components/MediaModal/MediaModal.test.tsx:12]: This is nice! I am slightly concerned with coupling these though (E.G breaking a unit test merely by changing the docs around) but perhaps it is unwarranted. Will leave it up to you.

File: src/components/MediaModal/MediaModal.tsx
Comment (juanfabrega) [inline src/components/MediaModal/MediaModal.tsx:116]: No reason to hold this up, but alternating between Box props and utility classes seems like something to keep an eye on. My preference is props simply because they are typed (or custom classes since we know those are only found in the `styles` module.

File: src/components/Modal/Modal.module.scss
Comment (juanfabrega) [reply src/components/Modal/Modal.module.scss:18]: I believe the components will throw a warning if the reach classes aren't included.

Comment (juanfabrega) [reply src/components/Modal/Modal.module.scss:18]: not 100% sure but because of css modules the class names get mangled and thus the warning is thrown.

**Review by juanfabrega — APPROVED** [review]
Hey @nathanyoung just left a few comments, but nothing crazy.

Some food for thought, I know that consolidating this with the Modal is likely not ideal for users of the library, but it might be worth it to consolidate the implementations under the hood (i.e: use the Modal as the underlying implementation of the MediaModal with some customization. It might save us down the line if the API changes so we don't have to carry the maintenance burden of two components that have matching APIs.

File: src/components/Modal/Modal.module.scss
Comment (nathanyoung) [reply src/components/Modal/Modal.module.scss:18]: this change doesn't generate any new warnings. there's another line in the css that you commented addresses this: https://github.com/palmetto/palmetto-components/blob/main/src/components/Modal/Modal.module.scss#L3-L4

File: src/components/MediaModal/MediaModal.tsx
Comment (nathanyoung) [reply src/components/MediaModal/MediaModal.tsx:116]: good point, will address.

File: src/components/Modal/Modal.module.scss
Comment (juanfabrega) [reply src/components/Modal/Modal.module.scss:18]: aah yes, forgot about that one. I think we could've removed the global classes at that point too.

**Pull request merged**
